### PR TITLE
RWA-2998 - Drop constraints in replica database

### PR DESCRIPTION
**Pull request by skandala-cgi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-2998

### Change description ###

Drop constraints in replica database to match to primary

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java [hunk 1]
```diff
@@ -435,6 +435,59 @@ void should_save_task_and_get_task_from_task_assignments() {
                 });
     }
 
+    @Test
+    void should_save_task_and_get_task_from_task_assignments_with_location_null() {
+        TaskResource taskResource = createAndAssignTask();
+        taskResource.setLocation(null);
```
Comment (kvvasudev) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java:441]: May be taskResource has to be saved again after update of location to null

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java [hunk 2]
```diff
@@ -435,6 +435,59 @@ void should_save_task_and_get_task_from_task_assignments() {
...
+
+                    return true;
+                });
+    }
+
+    @Test
+    void should_save_task_and_get_task_from_task_assignments_with_taskName_null() {
+        TaskResource taskResource = createAndAssignTask();
+        taskResource.setTaskName(null);
+        checkHistory(taskResource.getTaskId(), 1);
```
Comment (kvvasudev) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java:469]: Same as above comments for task_name too, after setting to null need to save taskResoource. Also need to add an assertion that the taskAssignement is created with task_name as null

File: src/main/resources/dbreplica/migration/V1.0.25__drop_unnecessary_constraint_replica.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE cft_task_db.task_assignments
```
Comment (BenNzewi) [inline src/main/resources/dbreplica/migration/V1.0.25__drop_unnecessary_constraint_replica.sql:1]: do we also need to address the assignee column?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java [hunk 4]
```diff
@@ -435,6 +437,61 @@ void should_save_task_and_get_task_from_task_assignments() {
...
+                    List<TaskAssignmentsResource> taskAssignmentsList
+                        = miReportingServiceForTest.findByAssignmentsTaskId(taskResource.getTaskId());
+
+                    assertFalse(taskAssignmentsList.isEmpty());
+                    assertEquals(1, taskAssignmentsList.size());
+                    assertEquals(taskResource.getTaskId(), taskAssignmentsList.get(0).getTaskId());
+                    assertEquals(taskResource.getTaskName(), taskAssignmentsList.get(0).getTaskName());
+                    assertEquals(taskResource.getAssignee(), taskAssignmentsList.get(0).getAssignee());
+                    assertEquals(taskResource.getJurisdiction(), taskAssignmentsList.get(0).getService());
+                    assertNull(taskAssignmentsList.get(0).getAssignmentEnd());
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java:461]: can update this after assignee column update to also verify assignee is Null.

**Pull request merged**